server/build_event_protocol/build_event_server/build_event_server.go

```diff
@@ -136,9 +136,16 @@ func (s *BuildEventProtocolServer) PublishBuildToolEventStream(stream pepb.Publi
 	for {
 		select {
 		case <-channelDone:
+			closeForwardingStreams(forwardingStreams)
 			return disconnectWithErr(status.FromContextError(channel.Context()))
 		case err := <-errCh:
+			closeForwardingStreams(forwardingStreams)
 			if err == io.EOF {
+				if s.synchronous {
```

Comment:
at face value it seems correct (and doesn't seem to be breaking anything) to wait here even though we've already gotten an EOF for the stream, but I was wondering if there was anything I should be worried about re: disconnecting + finalizing with an error when this stream has already effectively finished?

Reply:
Hm I'm not sure I understand the sequence you're describing - could you elaborate?